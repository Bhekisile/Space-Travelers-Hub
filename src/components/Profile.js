import React from 'react';
import { useSelector } from 'react-redux';
import './styles/Profile.css';

function Profile() {
  const joinedMissions = useSelector((state) => state.missions.joinedMissions);

  return (
    <div className="mission-container">
      <h2>My Missions</h2>
      <div className="joinedMission">
        {joinedMissions.length === 0 && <p className="no-mission">No mission joined</p>}
        {joinedMissions.length > 0 && (
          <ul>
            {joinedMissions.map((missionName) => (
              <li key={missionName.mission_id}>
                {missionName.mission_name}
              </li>
            ))}
          </ul>

        )}
      </div>
    </div>
  );
}

export default Profile;
