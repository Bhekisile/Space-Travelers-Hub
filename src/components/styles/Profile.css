.mission-container {
  margin-left: 100px;
  margin-top: 20px;
  width: 40%;
  min-height: 100px;
}

.joinedMission {
  margin-top: 5px;
}

ul {
  list-style: none;
  border: 1px solid #ccc;
}

li {
  height: 50px;
  border-bottom: 1px solid #ccc;
  padding-left: 15px;
  padding-top: 15px;
}

.no-mission {
  height: 30px;
  width: 90%;
  border: 1px solid #ccc;
  padding: 15px;
  margin-top: 10px;
}
