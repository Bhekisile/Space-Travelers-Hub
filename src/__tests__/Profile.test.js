import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../components/Profile';

const mockStore = configureStore([]);

test('Profile component matches snapshot', () => {
  const store = mockStore({
    missions: { joinedMissions: [] },
    rocket: [],
  });

  const { container } = render(
    <Provider store={store}>
      <Profile />
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
