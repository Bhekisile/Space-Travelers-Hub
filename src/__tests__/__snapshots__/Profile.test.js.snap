// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component matches snapshot 1`] = `
<div>
  <div
    class="profile-container"
  >
    <div
      class="mission-container"
    >
      <h2>
        My Missions
      </h2>
      <div
        class="joinedMission"
      >
        <p
          class="no-mission"
        >
          No mission joined
        </p>
      </div>
    </div>
    <div
      class="Rockets"
    >
      <h2
        class="head"
      >
        My Rockets
      </h2>
      <p
        class="no-mission"
      >
        You haven't reserved any rockets yet.
      </p>
    </div>
  </div>
</div>
`;
